P1754R1 Rename concepts to standard_case for C++20

burblebee opened the pull request.

CaseyCarter commented on source/utilities.tex:
```diff
@@ -8221,11 +8221,11 @@
-      requires DefaultConstructible<iter_value_t<I>>
-    I uninitialized_default_construct(I first, S last);
+    requires default_constructible<iter_value_t<I>>
+      I uninitialized_default_construct(I first, S last);
```
This is definitely the style we used for the majority of the declarations with *requires-clause*s introduced by P0896R4; apparently these were missed.

CaseyCarter commented on source/algorithms.tex:
```diff
@@ -1434,33 +1434,35 @@
-    template<InputIterator I, Sentinel<I> S, class T1, class T2, OutputIterator<const T2&> O,
+    template<input_iterator I, sentinel_for<I> S,
+             class T1, class T2, output_iterator<const T2&> O,
              class Proj = identity>
```
Any reason not to two-line this *template-head* a la:
```c++
template<input_iterator I, sentinel_for<I> S, class T1, class T2, 
         output_iterator<const T2&> O, class Proj = identity>
```

burblebee replied:
Was trying to create some consistency with how other params were wrapped, but I like your suggestion - fixed.

CaseyCarter commented on source/algorithms.tex:
```diff
@@ -1650,20 +1652,20 @@
+             class Proj = identity, indirect_relation<projected<I, Proj>> C = ranges::equal_to>
+      requires indirectly_copyable<I, O> &&
+               (forward_iterator<I> ||
+                (input_iterator<O> && same_to<iter_value_t<I>, iter_value_t<O>>) ||
```
```suggestion
                (input_iterator<O> && same_as<iter_value_t<I>, iter_value_t<O>>) ||
```

CaseyCarter replied:
and on 1667.

burblebee replied:
I've done a global change for this, thanks.

burblebee commented on source/numerics.tex:
```diff
@@ -1947,17 +1947,17 @@
-      { G::min() } -> Same<invoke_result_t<G&>>;
-      { G::max() } -> Same<invoke_result_t<G&>>;
+      { G::min() } -> same_to<invoke_result_t<G&>>;
+      { G::max() } -> same_to<invoke_result_t<G&>>;
```
Doh!  not sure where same_to came from :)

burblebee commented on source/algorithms.tex:
```diff
@@ -808,16 +808,16 @@
-             IndirectRelation<projected<I1, Proj1>,
-                              projected<I2, Proj2>> Pred = ranges::equal_to>
+             indirect_relation<projected<I1, Proj1>,
+                                projected<I2, Proj2>> Pred = ranges::equal_to>
```
thanks for catching these!

burblebee commented on source/algorithms.tex:
```diff
@@ -3848,8 +3854,8 @@
 \begin{itemize}
-\item \tcode{S1} and \tcode{I1} model \libconcept{SizedSentinel},
-\item \tcode{S2} and \tcode{I2} model \libconcept{SizedSentinel}, and
+\item \tcode{S1} and \tcode{I1} model \libconcept{sized_sentinel_for},
```
This isn't part of the paper, but I went ahead and made this change in a separate commit.  It's a general problem that needs to be fixed throughout the wording - please open an issue for the general case.  Thanks!

johelegp commented on source/concepts.tex:
```diff
@@ -269,21 +269,21 @@
 
 template<class T, class U>
-  concept same_to = @\placeholdernc{same-impl}@<T, U> && @\placeholdernc{same-impl}@<U, T>;
+  concept same_as = @\placeholdernc{same-impl}@<T, U> && @\placeholdernc{same-impl}@<U, T>;
```
It seems appropriate to rename _`same-impl`_ to _`same-as-impl`_.

burblebee replied:
Combined this with a couple other placeholder name changes mentioned in #3015 as a separate commit.

CaseyCarter commented:
Thanks again for all the work on this, @burblebee, I really should have prepared this PR myself.

burblebee commented:
@jwakely you marked this "changes requested" - I addressed all of your concerns afaik.  Might you please have a quick look to make that go away?  I don't like the red :)

burblebee commented:
@zygoloid this motion is not complete - must be completed after all other motions are merged.

burblebee commented on source/iterators.tex:
```diff
@@ -1121,13 +1121,13 @@
-\tcode{E2} model \tcode{IndirectlyMovableStorable<T1, T2>} and
-\tcode{IndirectlyMovableStorable<T2, T1>}, then
+\tcode{E2} model \tcode{\libconcept{indirectly_movable_storable}<T1, T2>} and
+\tcode{\libconcept{indirectly_movable_storable}<T2, T1>}, then
```
Don't know, but it seemed most existing Latex used \tcode{\libconcept{foo}<T1, T2>}, so I went with that for consistency.

CaseyCarter replied:
I used `\tcode{\libconcept{C}<T, U>}` in the existing wording for consistency with `\tcode{otherstuff && \libconcept{C}<T, U>}`; `\tcode{otherstuff && }\libconcept{C}\tcode{<T, U>}` feels "weird" to me.

zygoloid commented:
Rebasing this turned out to be more work than recreating the semi-mechanical renaming part, so I did that and kept all the other changes. This is superseded by https://github.com/cplusplus/draft/pull/3149.
